tap_deputy/client.py

```diff
@@ -19,17 +24,55 @@ def __init__(self, config, config_path):
         self.__client_secret = config.get('client_secret')
         self.__redirect_uri = config.get('redirect_uri')
         self.__refresh_token = config.get('refresh_token')
-        self.__access_token = None
-        self.__expires_at = None
+        self.__access_token = config.get('access_token')
         self.__session = requests.Session()
+        self.__dev_mode = dev_mode
+        self.__expires_at = strptime_to_utc(config.get('expires_at')) \
+            if config.get("expires_at") else None
+
+    @property
+    def config(self):
+        return self.__config
+
+    @property
+    def refresh_token(self):
+        return self.__refresh_token
+
+    @property
+    def access_token(self):
+        return self.__access_token
+
+    @property
+    def expires_at(self):
+        return self.__expires_at
 
     def __enter__(self):
         return self
 
-    def __exit__(self, type, value, traceback):
+    def __exit__(self, _type, value, traceback):
         self.__session.close()
 
     def refresh(self):
+        """
+        Checks token expiry and refresh token when it is expired and dev mode is not enabled
+        """
+        if self.__dev_mode:
+            if not self.__access_token:
+                raise Exception("Access token config property is missing")
+
+            if not self.__expires_at:
+                raise Exception(
+                    "Expiry of access token config property is missing")
+
+            if self.__expires_at < now():
+                raise Exception(
```

Comment:
I don't think the expires_at values are necessary to be this strict for dev mode. If you're running it in dev mode and the token is expired, it should just fail.

Reply:
This conditional can just return after the check for `access_token`, since that's necessary to make a request in the first place.

Reply:
Updated the implementation.